File: fern/snippets/tool-calling/python/auth-tools.py
from composio import Composio
from composio.types import auth_scheme

linear_auth_config_id = "ac_dqYN9oElNVlg"
user_id = "0000-1111-2222"
composio = Composio()

# Create a new connected account
connection_request = composio.connected_accounts.initiate(
    user_id=user_id,
    auth_config_id=linear_auth_config_id,
    config=auth_scheme.oauth2()
)
print(connection_request.redirect_url)

# Wait for the connection to be established
connected_account = connection_request.wait_for_connection()

# If you only have the connection request ID, you can also wait using:

connected_account = composio.connected_accounts.wait_for_connection(connection_request.id)
# Recommended for when connection_request object is destroyed

# API key based toolkit
serp_auth_config_id = "ac_VWmFEC55Zgv6"

# Retrieved from the user
user_api_key = "sk_1234567890"

connection_request = composio.connected_accounts.initiate(
    user_id=user_id,
    auth_config_id=serp_auth_config_id,
    config=auth_scheme.api_key(user_api_key)
)

File: fern/snippets/tool-calling/typescript/auth-tools.ts
import { Composio } from '@composio/core';

const composio = new Composio();
const linearAuthConfigId = 'ac_dqYN9oElNVlg';

const userId = '1111-2222-3333';

// Initiate the OAuth connection request
const connRequest = await composio.connectedAccounts.initiate(userId, linearAuthConfigId);

// Destructure redirectUrl for easier access
const { redirectUrl, id } = connRequest;
console.log(redirectUrl);

// Wait for the connection to be established
await connRequest.waitForConnection();

// If you only have the connection request ID, you can also wait using:
// Recommended for when connRequest object is destroyed
await composio.connectedAccounts.waitForConnection(id);

// API Key based toolkit

import { AuthScheme } from '@composio/core';
const serpAuthConfigId = 'ac_VWmFEC55Zgv6';

// Retrieved from the user
const userApiKey = 'sk_1234567890';

const serpConnectionRequest = await composio.connectedAccounts.initiate(userId, serpAuthConfigId, {
  config: AuthScheme.APIKey({
    api_key: userApiKey,
  }),
});

console.log(JSON.stringify(serpConnectionRequest, null, 2));

// Auth scheme for Airtable Bearer
import { AuthScheme } from '@composio/core';
const airtableAuthConfigId = 'ac_1234567890';

// Retrieved from the user
const userBearerToken = '1234567890';

const airtableConnectionRequest = await composio.connectedAccounts.initiate(
  userId, airtableAuthConfigId, {
  config: AuthScheme.BearerToken({
    token: userBearerToken,
  }),
});
